hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,570 @@
+
+from odoo.http import request, Response, Controller, route
+import json
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class MainController(Controller):
+    @route('/create_contact', methods=['POST'], type='json', auth='user' )
+    def create_contact(self):
+        
+        # Lista de todos los campos posibles
+
+        all_fields = [
+            'company_type', 'name', 'phone', 'street_number', 'street_number2',
+            'email', 'street', 'street2', 'city', 'zip', 'state_id', 'country_id',
+            'vat', 'l10n_mx_edi_curp', 'website', 'lang', 'category_id', 'l10n_mx_edi_fiscal_regime', 'parent_id'
+        ]
+
+        # Construir el diccionario contact_data con solo los campos enviados en el JSON
+        contact_data = {field: request.jsonrequest.get(field) for field in all_fields if request.jsonrequest.get(field) is not None}
+
+        # Si category_id está presente, convertirlo en una lista
+        if 'category_id' in contact_data and not isinstance(contact_data['category_id'], list):
+            contact_data['category_id'] = [contact_data['category_id']]
+        
+        list_not_title = ['lang', 'company_type', 'email', 'website']
+
+        for key, value in contact_data.items():
+            if isinstance(value, str) and key not in list_not_title:
+                    contact_data[key] = value.title()
```

Comment:
why did you change the values?

you should register as it comes, not data transforming